gpytorch/priors/lkj_prior.py

```diff
@@ -89,6 +130,10 @@ def log_prob(self, X):
         log_diag_sum = torch.diagonal(X, dim1=-2, dim2=-1).log().sum(-1)
         return self.C + (self.eta - 1) * 2 * log_diag_sum
 
+    def rsample(self, sample_shape):
+        res = super()._rsample(sample_shape=sample_shape, return_covariance=False)
+        return res.cholesky()
```

Comment:
This seems like unnecessary work. IIUC in `_rsample` of `LKJPrior` you're essentially sampling an upper triangular Cholesky factor `R`. Can you just make a `_rsample_cholesky` method that does that (and then call this here), and then have the forming of the covariance/correlation matrix happen in `LKJPrior.rsample`?

Reply:
yeah, the issue is that the cholesky returned is the cholesky of the covariance matrix we're sampling, not the correlation matrix corresponding to that covariance. off the top of my head, i dont think there's a way to re-use the cholesky under that type of transformation.